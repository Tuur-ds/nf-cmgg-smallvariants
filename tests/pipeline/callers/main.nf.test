nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_callers"

    options "-stub"

    test("pipeline_callers - vardict") {

        when {
            params {
                outdir = "${outputDir}"
                callers = "vardict"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("vardict") }
            )
        }

    }

    test("pipeline_callers - haplotypecaller") {

        when {
            params {
                outdir = "${outputDir}"
                callers = "haplotypecaller"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("haplotypecaller") }
            )
        }

    }

    test("pipeline_callers - vardict + haplotypecaller") {

        when {
            params {
                outdir = "${outputDir}"
                callers = "haplotypecaller,vardict"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("vardict + haplotypecaller") }
            )
        }

    }

}

def getRecursiveFileNames(fileOrDir, outputDir) {
    if(file(fileOrDir.toString()).isDirectory()) {
        return fileOrDir.list().collect { getRecursiveFileNames(it, outputDir) }
    }
    return fileOrDir.toString().replace("${outputDir}/", "")
}
