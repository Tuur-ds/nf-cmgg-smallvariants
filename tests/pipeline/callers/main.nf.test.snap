{
    "haplotypecaller": {
        "content": [
            [
                
            ],
            [
                "NA24143/NA24143.bed",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "NA24143/reports/NA24143.haplotypecaller.bcftools_stats.txt",
                "NA24149/NA24149.bed",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "NA24149/reports/NA24149.haplotypecaller.bcftools_stats.txt",
                "NA24385/NA24385.bed",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "NA24385/reports/NA24385.haplotypecaller.bcftools_stats.txt",
                "multiqc/multiqc_report.html",
                "samplesheet.csv",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.bed",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.ped",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T15:43:23.245304993"
    },
    "vardict + haplotypecaller": {
        "content": [
            [
                
            ],
            [
                "NA24143/NA24143.bed",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz",
                "NA24143/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "NA24143/reports/NA24143.haplotypecaller.bcftools_stats.txt",
                "NA24149/NA24149.bed",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz",
                "NA24149/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "NA24149/reports/NA24149.haplotypecaller.bcftools_stats.txt",
                "NA24385/NA24385.bed",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz",
                "NA24385/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "NA24385/reports/NA24385.haplotypecaller.bcftools_stats.txt",
                "multiqc/multiqc_report.html",
                "samplesheet.csv",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.bed",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.ped",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "test/Ashkenazim/NA24143.vardict.ped",
                "test/Ashkenazim/NA24143.vardict.vcf.gz",
                "test/Ashkenazim/NA24143.vardict.vcf.gz.tbi",
                "test/Ashkenazim/NA24149.vardict.ped",
                "test/Ashkenazim/NA24149.vardict.vcf.gz",
                "test/Ashkenazim/NA24149.vardict.vcf.gz.tbi",
                "test/Ashkenazim/NA24385.vardict.ped",
                "test/Ashkenazim/NA24385.vardict.vcf.gz",
                "test/Ashkenazim/NA24385.vardict.vcf.gz.tbi",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.html",
                "test/Ashkenazim/reports/NA24143.vardict.bcftools_stats.txt",
                "test/Ashkenazim/reports/NA24143.vardict.html",
                "test/Ashkenazim/reports/NA24149.vardict.bcftools_stats.txt",
                "test/Ashkenazim/reports/NA24149.vardict.html",
                "test/Ashkenazim/reports/NA24385.vardict.bcftools_stats.txt",
                "test/Ashkenazim/reports/NA24385.vardict.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T15:44:48.221710874"
    },
    "vardict": {
        "content": [
            [
                
            ],
            [
                "NA24143/NA24143.bed",
                "NA24149/NA24149.bed",
                "NA24385/NA24385.bed",
                "multiqc/multiqc_report.html",
                "samplesheet.csv",
                "test/Ashkenazim/NA24143.vardict.ped",
                "test/Ashkenazim/NA24143.vardict.vcf.gz",
                "test/Ashkenazim/NA24143.vardict.vcf.gz.tbi",
                "test/Ashkenazim/NA24149.vardict.ped",
                "test/Ashkenazim/NA24149.vardict.vcf.gz",
                "test/Ashkenazim/NA24149.vardict.vcf.gz.tbi",
                "test/Ashkenazim/NA24385.vardict.ped",
                "test/Ashkenazim/NA24385.vardict.vcf.gz",
                "test/Ashkenazim/NA24385.vardict.vcf.gz.tbi",
                "test/Ashkenazim/reports/NA24143.vardict.bcftools_stats.txt",
                "test/Ashkenazim/reports/NA24143.vardict.html",
                "test/Ashkenazim/reports/NA24149.vardict.bcftools_stats.txt",
                "test/Ashkenazim/reports/NA24149.vardict.html",
                "test/Ashkenazim/reports/NA24385.vardict.bcftools_stats.txt",
                "test/Ashkenazim/reports/NA24385.vardict.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T15:42:12.627820502"
    }
}